python_modules/dagster-pipes/dagster_pipes/__init__.py

```diff
@@ -550,6 +587,18 @@ def get_opened_extras(self) -> PipesExtras:
         """
         return {}
 
+    @property
+    def has_log_writer(self) -> bool:
+        return hasattr(self, "_log_writer") and self._log_writer is not None
+
+    @property
+    def log_writer(self) -> "PipesLogWriter":
+        if not self.has_log_writer or self._log_writer is None:
+            raise DagsterPipesError(
+                "The PipesMessageWriter does not have a PipesLogWriter attached."
+            )
+        return self._log_writer
+
```

Comment:
Not clear to me why we are defining this logic in `PipesMessageReader` when we only ever set a log writer in `PipesBlobStoreMessageWriter`. Seems we shouldn't need to touch `PipesMessageWriter` at all, we can just move all the logic down into `PipesBlobStoreMessageWriter` and override `get_opened_extras` instead of rewriting `get_opened_payload`.

Reply:
I have a feeling you might be confusing `PipesLogWriter` with `PipesDefaultLogWriter`. 

`PipesLogWriter` in general can be used with any `PipesMesssageWriter` (in fact, they are independent and we are only nesting them to copy this nesting structure on the orchestration side). 

`PipesDefaultLogWriter` is a concrete implementation of `PipesLogWriter` which writes logs via Pipes messages. 

We decided that for now we are only going to expose `PipesDefaultLogWriter` as an argument to `PipesBlobStoreMessageWriter`, but it doesn't mean that all `PipesLogWriter` logic has to go inside `PipesBlobStoreMessageWriter`. 

Am I missing something?